docs/middleware/callbacks/integration.md

```diff
@@ -0,0 +1,98 @@
+* [IBC middleware integration](../../ibc/middleware/integration.md) {prereq}
+
+The callbacks middleware, as the name suggests, plays the role of an IBC middleware and as such must be configured by chain developers to route and handle IBC messages correctly.
+For Cosmos SDK chains this setup is done via the `app/app.go` file, where modules are constructed and configured in order to bootstrap the blockchain application.
+
+## Configuring an application stack with the callbacks middleware
+
+As mentioned in [IBC middleware development](../../ibc/middleware/develop.md) an application stack may be composed of many or no middlewares that nest a base application.
+These layers form the complete set of application logic that enable developers to build composable and flexible IBC application stacks.
+For example, an application stack may be just be a single base application like `transfer`, however, the same application stack composed with `29-fee` and `callbacks` will nest the `transfer` base application twice by wrapping it with the Fee Middleware module and then callbacks middleware.
+
+The callbacks middleware also **requires** a secondary application that will receive the callbacks to implement the [`ContractKeeper`](https://github.com/cosmos/ibc-go/blob/main/modules/apps/callbacks/types/expected_keepers.go#L11-L64). Since the wasm module does not yet support the callbacks middleware, we will use the mockContractKeeper module in the examples below. You should replace this with your own module that implements `ContractKeeper`.`
+
+### Transfer
+
+See below for an example of how to create an application stack using `transfer`, `29-fee`, and `callbacks`. Feel free to omit the `29-fee` middleware if you do not want to use it.
+The following `transferStack` is configured in `app/app.go` and added to the IBC `Router`.
+The in-line comments describe the execution flow of packets between the application stack and IBC core.
+
+```go
+// Create Transfer Stack
+// SendPacket, since it is originating from the application to core IBC:
+// transferKeeper.SendPacket -> callbacks.SendPacket -> fee.SendPacket -> channel.SendPacket
+
+// RecvPacket, message that originates from core IBC and goes down to app, the flow is the other way
+// channel.RecvPacket -> callbacks.OnRecvPacket -> fee.OnRecvPacket -> transfer.OnRecvPacket
+
+// transfer stack contains (from top to bottom):
+// - IBC Callbacks Middleware
+// - IBC Fee Middleware
+// - Transfer
+
+// create IBC module from bottom to top of stack
+var transferStack porttypes.IBCModule
+transferStack = transfer.NewIBCModule(app.TransferKeeper)
+transferStack = ibcfee.NewIBCMiddleware(transferStack, app.IBCFeeKeeper)
+transferStack = ibccallbacks.NewIBCMiddleware(transferStack, app.IBCFeeKeeper, app.MockContractKeeper, maxCallbackGas)
+// Since the callbacks middleware itself is an ics4wrapper, it needs to be passed to the transfer keeper
+app.TransferKeeper.WithICS4Wrapper(transferStack.(porttypes.Middleware))
```

Comment:
I don't understand what you mean

Reply:
I was just wondering if it would be possible to cast `transferStack` to `porttypes.ICS4Wrapper` instead of `porttypes.Middleware`. I thought that maybe casting to `porttypes.ICS4Wrapper` would be enough.

Reply:
Ok. Since this will touch the code. Should I do it in another PR which both updates the code and the doc?